cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -1123,6 +1123,63 @@ public void deprecatePackage(String packageInfo, String deprecationMsg, String s
         }
     }
 
+    /**
+     * Get packages from central.
+     *
+     * @param params            Search query param map.
+     * @param supportedPlatform The supported platform.
+     * @param ballerinaVersion  The ballerina version.
+     * @return Package list
+     * @throws CentralClientException Central client exception.
+     */
+    public JsonElement getPackages(Map<String, String> params, String supportedPlatform, String ballerinaVersion)
+            throws CentralClientException {
+        Optional<ResponseBody> body = Optional.empty();
+        OkHttpClient client = new OkHttpClient.Builder()
+                .followRedirects(false)
+                .connectTimeout(20, TimeUnit.SECONDS)
+                .readTimeout(20, TimeUnit.SECONDS)
+                .proxy(this.proxy)
+                .build();
+
+        try {
+            HttpUrl.Builder httpBuilder = HttpUrl.parse(this.baseUrl).newBuilder().addPathSegment(PACKAGES);
+            for (Map.Entry<String, String> param : params.entrySet()) {
+                httpBuilder.addQueryParameter(param.getKey(), param.getValue());
+            }
+
+            Request searchReq = getNewRequest(supportedPlatform, ballerinaVersion)
+                    .get()
+                    .url(httpBuilder.build())
+                    .build();
+
+            Call httpRequestCall = client.newCall(searchReq);
+            Response searchResponse = httpRequestCall.execute();
+
+            ResponseBody responseBody = searchResponse.body();
+            body = responseBody != null ? Optional.of(responseBody) : Optional.empty();
```

Comment:
Why do we need a new variable called `body`

Reply:
We need it in https://github.com/ballerina-platform/ballerina-lang/pull/40502/files#diff-799a37ade2d32a3824ea488bb9ed825de0ddf34b403c9b522a7197664b75b51dR1173